## Course overview
Analysis of genetic content of microbial communities through metagenomics has become the mainstream methodology.
SOMETHING ABOUT ENVIRONMENTs AND THEIR COMPLEXITY IN HERE. MAYBE ALSO ABOUT ADVANTAGES OF METAGENOMICS COMPARED TO AMPLICONS.
During the one week course the students will learn and apply bioinformatic techniques to study microbial communities with metagenomic approaches.
The students will learn both read-based and assembly-based methods to analyze metagenomic data. We will use data from both short and long read sequencing platforms.
The goal is to become familiar with the bioinformatic analysis tools and to be able to utilize them in own research after the course.

## Learning outcomes
- The students understand the basics of metagenomic sequencing and data analyis and are able to participate in planning a metagenomic sequencing project
- The students can assess the different bioinformatic methods for metagenomic data analysis and choose the right tools for their own data
- The students will be able to analyse their own metagenomic data and have confidence to learn new methods needed to answer their research questions

## Program overview

### Monday
- High-throughput sequencing technologies
- Repositories for high throughput sequencing data
- Data download
- QC and trimming

### Tuesday
- Taxonomic annotation of metagenomes
- Functional annotation of metagenomes

### Wednesday
- Metagenome assembly
- Hybrid assembly
- Assembly QC

### Thursday
- Genome-resolved metagenomics (Anvi'o)

### Friday
- Annotation of MAGs
